presto-main/src/main/java/com/facebook/presto/operator/scalar/MergingDigest.java

```diff
@@ -35,64 +41,45 @@
  * N/k + log k
  * <p>
  * These costs even out when N/k = log k.  Balancing costs is often a good place to start in optimizing an
- * algorithm.  For different values of compression factor, the following table shows estimated asymptotic
- * values of N and suggested values of k:
- * <table>
- * <thead>
- * <tr><td>Compression</td><td>N</td><td>k</td></tr>
- * </thead>
- * <tbody>
- * <tr><td>50</td><td>78</td><td>25</td></tr>
- * <tr><td>100</td><td>157</td><td>42</td></tr>
- * <tr><td>200</td><td>314</td><td>73</td></tr>
- * </tbody>
- * <caption>Sizing considerations for t-digest</caption>
- * </table>
- * <p>
+ * algorithm.
  * The virtues of this kind of t-digest implementation include:
  * <ul>
  * <li>No allocation is required after initialization</li>
  * <li>The data structure automatically compresses existing centroids when possible</li>
  * <li>No Java object overhead is incurred for centroids since data is kept in primitive arrays</li>
  * </ul>
  * <p>
- * The current implementation takes the liberty of using ping-pong buffers for implementing the merge resulting
- * in a substantial memory penalty, but the complexity of an in place merge was not considered as worthwhile
- * since even with the overhead, the memory cost is less than 40 bytes per centroid which is much less than half
- * what the AVLTreeDigest uses and no dynamic allocation is required at all.
  */
-public class MergingDigest extends AbstractTDigest {
-    private int mergeCount = 0;
+
+public class MergingDigest
+        extends TDigest
+{
+    private int mergeCount;
 
     private final double publicCompression;
     private final double compression;
 
     // points to the first unused centroid
-    private int lastUsedCell;
+    private int activeCentroids;
 
-    // sum_i weight[i]  See also unmergedWeight
-    private double totalWeight = 0;
+    private double totalWeight;
 
-    // number of points that have been added to each merged centroid
     private final double[] weight;
-    // mean of points added to each merged centroid
+
     private final double[] mean;
 
-    // history of all data added to centroids (for testing purposes)
```

Comment:
Why do we remove all the comments?

Reply:
I felt like it was clear from the code what that array represents, so no need to add comments.